source/extensions/all_extensions.bzl

```diff
@@ -1,31 +1,41 @@
+load("@envoy_build_config//:extensions_build_config.bzl", "EXTENSIONS")
+
 # Return all extensions to be compiled into Envoy.
-# TODO(mattklein123): Every extension should have an independent Bazel select option that will
-# allow us to compile in and out different extensions. We may also consider in the future other
-# selection options such as maturity.
-def envoy_all_extensions(repository = ""):
-  return [
-    repository + "//source/extensions/access_loggers/file:config",
-    repository + "//source/extensions/access_loggers/http_grpc:config",
-    repository + "//source/extensions/filters/http/dynamo:config",
-    repository + "//source/extensions/filters/http/ext_authz:config",
-    repository + "//source/extensions/filters/http/lua:config",
-    repository + "//source/extensions/filters/http/ratelimit:config",
-    repository + "//source/extensions/filters/listener/proxy_protocol:config",
-    repository + "//source/extensions/filters/listener/original_dst:config",
-    repository + "//source/extensions/filters/network/client_ssl_auth:config",
-    repository + "//source/extensions/filters/network/echo:config",
-    repository + "//source/extensions/filters/network/ext_authz:config",
-    repository + "//source/extensions/filters/network/mongo_proxy:config",
-    repository + "//source/extensions/filters/network/ratelimit:config",
-    repository + "//source/extensions/filters/network/redis_proxy:config",
-    repository + "//source/extensions/filters/network/tcp_proxy:config",
-    repository + "//source/extensions/stat_sinks/dog_statsd:config",
-    repository + "//source/extensions/stat_sinks/metrics_service:config",
-    repository + "//source/extensions/stat_sinks/statsd:config",
-    repository + "//source/extensions/tracers/dynamic_ot:config",
-    repository + "//source/extensions/tracers/lightstep:config",
-    repository + "//source/extensions/tracers/zipkin:config",
-    repository + "//source/extensions/transport_sockets/raw_buffer:config",
-    repository + "//source/extensions/transport_sockets/ssl:config",
+# TODO(mattklein123): Figure out a way to output in the build and possibly even the binary which
+#                     extensions are statically registered.
```

Comment:
Is the idea we would create custom ELF headers for this?

Reply:
I was thinking more like a CLI flag that enumerates the various registries and dumps all statically registered things. I will make the TODO more detailed.